pkg/adapter/rule.go

```diff
@@ -39,7 +40,11 @@ func addSuffix(ref, suffix string) (string, error) {
 		return "", errors.Wrap(err, "invalid source image reference")
 	}
 	if _, ok := named.(docker.Digested); ok {
-		return "", fmt.Errorf("unsupported digested image reference: %s", named.String())
+		re := regexp.MustCompile(`@sha256.*`)
```

Comment:
Better to handle with `named` instead of regexp.

Reply:
replace by `docker.TrimNamed(named)`.